reactor-core/src/test/java/reactor/core/publisher/FluxPublishTest.java

```diff
@@ -720,4 +721,28 @@ public void removeUnknownInnerIgnored() {
 		subscriber.remove(inner);
 		assertThat(subscriber.subscribers).as("post remove inner").isEmpty();
 	}
+
+	@Test
+	public void subscriberContextPropagation() {
+		String key = "key";
+		int expectedValue = 1;
+
+		Flux<Integer> integerFlux =
+				Flux.just(1, 2, 3)
+				    .flatMap(value ->
+						    Mono.subscriberContext()
+						        .doOnNext(context -> assertThat((int) context.get(key))
```

Comment:
To avoid false positives, could you please make an assertion outside of the reactive chain (e.g. by capturing the context and doing the `assertThat` call after `subscribe()`s?

Reply:
Is it ok using an `AtomicReference`?